auth/api/iam/api.go

```diff
@@ -888,25 +929,13 @@ func (r Wrapper) authzRequestObjectStore() storage.SessionStore {
 	return r.storageEngine.GetSessionDatabase().GetStore(accessTokenValidity, oauthRequestObjectKey...)
 }
 
-// createOAuth2BaseURL creates an OAuth2 base URL for an owned did:web DID
-// It creates a URL in the following format: https://<did:web host>/<base path>/oauth2/<did>
-func createOAuth2BaseURL(webDID did.DID) (*url.URL, error) {
-	didURL, err := didweb.DIDToURL(webDID)
-	if err != nil {
-		return nil, fmt.Errorf("failed to convert DID to URL: %w", err)
-	}
-	// Part until /iam/<webDID> is the base path, which we need to prepend
-	tenantIdx := strings.Index(didURL.Path, "/iam/")
-	var basePath string
-	if tenantIdx != -1 {
-		basePath = didURL.Path[:tenantIdx]
-	}
-	result, err := didURL.Parse("/")
-	if err != nil {
-		return nil, err
+func (r Wrapper) subjectToBaseURL(subject string) url.URL {
+	u := &url.URL{}
+	publicURL := r.auth.PublicURL()
+	if publicURL != nil {
```

Comment:
It's mandatory, so can't be nil

Reply:
yes, but the road from core serverConfig to place of use is long and there are many tests that ignore it (but do call it). Changed it to a panic for the API.